test/ATokenYieldSource.test.ts

```diff
@@ -97,7 +98,9 @@ describe('ATokenYieldSource', () => {
         hardhatATokenYieldSourceHarness.address,
         contractsOwner,
       )) as ATokenYieldSourceHarness;
-  
+    
+    await underlyingToken.mock.safeApprove.returns(true);
```

Comment:
To mock `safeApprove`, you actually need to mock `approve`, so something like:
```suggestion
    await underlyingToken.mock.approve.withArgs(lendingPoolAddress, maxAmount).returns(true);
```